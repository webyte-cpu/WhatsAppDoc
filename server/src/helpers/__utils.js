import objectKeysToCamelCase from "./objectKeyCase.js";
import objectFilter from "./objectFilter.js";
import { constructTestServer, cleanDb ,} from "./integrationTesting.js";

export {
  constructTestServer,
  objectKeysToCamelCase,
  objectFilter,
  cleanDb,
};
