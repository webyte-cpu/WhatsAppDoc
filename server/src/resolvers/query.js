import admin from "../models/admin.js";
import doctor from "../models/doctor.js";
import patient from "../models/patient.js";
import user from "../models/user.js";

const query = {
  admin: () => admin.get(),
  user: () => user.get(),
  doctor: () => doctor.get(),
  // patient: () => patient.get(),
};

export default query;
