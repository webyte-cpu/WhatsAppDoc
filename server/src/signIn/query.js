import { gql } from "apollo-server-express";

export const SIGN_IN = gql`
  mutation SignIn($email: EmailAddress!, $password: String!) {
    signIn(email: $email, password: $password)
  }
`;
