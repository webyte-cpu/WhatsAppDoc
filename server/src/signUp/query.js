import { gql } from "apollo-server-express";

export const SIGN_UP = gql`
  mutation SignUp(
    $firstName: String!
    $middleName: String
    $lastName: String!
    $email: EmailAddress!
    $password: String!
    $role: Role!
    $img: String
    $sex: Sex!
    $birthdate: Date!
    $doctor: DoctorInput
  ) {
    signUp(
      firstName: $firstName
      middleName: $middleName
      lastName: $lastName
      email: $email
      password: $password
      role: $role
      img: $img
      sex: $sex
      birthdate: $birthdate
      doctor: $doctor
    )
  }
`;
