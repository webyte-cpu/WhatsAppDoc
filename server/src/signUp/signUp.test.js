import { cleanDb, constructTestServer } from "../helpers/__utils.js";
import { createTestClient } from "apollo-server-testing";
import jwt from "jsonwebtoken";
import { SIGN_UP } from "./query.js";
import pg from "../../db/index.js";
describe("Sign Up", () => {
  beforeAll(async () => {
    try {
      await cleanDb();
      await pg.seed.run();
    } catch (error) {
      console.log(error);
    }
  });

  afterAll(async () => {
    try {
      await cleanDb();
      await pg.destroy();
    } catch (error) {
      console.log(error);
    }
  });

  const { server } = constructTestServer();
  const { query } = createTestClient(server);

  it("Has an invalid email and password", async () => {
    const variables = {
      email: "",
      password: "",
      firstName: "Kyle",
      middleName: null,
      lastName: "Osunero",
      role: "PATIENT",
      img: "image Path",
      sex: "MALE",
      birthdate: "2000-07-02",
    };

    const response = await query({
      mutation: SIGN_UP,
      variables,
    });

    await expect(response).toMatchSnapshot();
  });

  it("Has valid email but invalid password", async () => {
    const variables = {
      email: "wex@webyte.org",
      password: "asd",
      firstName: "wex",
      middleName: null,
      lastName: "dalisay",
      role: "DOCTOR",
      img: "image Path",
      sex: "FEMALE",
      birthdate: "2000-09-02",
      doctor: {
        licenceNum: "12345678",
        licenceImg: "img path",
        licenceExp: "2100-01-01",
        specialization: ["special powers", "healing powers"],
      },
    };

    const response = await query({
      mutation: SIGN_UP,
      variables,
    });

    await expect(response).toMatchSnapshot();
  });

  it("Trys to sign up with existing user", async () => {
    const variables = {
      email: "Billie97@yahoo.com",
      password: "L3tM3!N!!",
      firstName: "Kent",
      middleName: null,
      lastName: "Handumon",
      role: "PATIENT",
      img: "image Path",
      sex: "MALE",
      birthdate: "1999-03-02",
    };

    const response = await query({
      mutation: SIGN_UP,
      variables,
    });
    await expect(response).toMatchSnapshot();
  });

  it("Should sign up a Doctor", async () => {
    const variables = {
      email: "doc_kyle@webyte.org",
      password: "$3cR3tP4ss2",
      firstName: "Kyle",
      middleName: null,
      lastName: "Osunero",
      role: "DOCTOR",
      img: "image Path",
      sex: "MALE",
      birthdate: "2000-05-02",
      doctor: {
        licenceNum: "12345678",
        licenceImg: "img path",
        licenceExp: "2100-10-10",
        specialization: ["special powers", "healing powers"],
      },
    };

    const response = await query({
      mutation: SIGN_UP,
      variables,
    });

    const { email, firstName, lastName, role, sex, img, birthdate } =
      jwt.verify(response.data.signUp, process.env.JWT_SECRET_KEY);

    await expect({
      email,
      firstName,
      lastName,
      role,
      sex,
      img,
      birthdate,
    }).toMatchSnapshot();
  });

  it("Should sign up a Patient", async () => {
    const variables = {
      email: "patient_kyle@webyte.org",
      password: "$3cR3tP4ss2",
      firstName: "Kyle",
      middleName: null,
      lastName: "Osunero",
      role: "PATIENT",
      img: "image Path",
      sex: "MALE",
      birthdate: "2000-08-02",
    };

    const response = await query({
      mutation: SIGN_UP,
      variables,
    });

    console.log(response);

    const { email, firstName, lastName, role, sex, img, birthdate } =
      jwt.verify(response.data.signUp, process.env.JWT_SECRET_KEY);

    await expect({
      email,
      firstName,
      lastName,
      role,
      sex,
      img,
      birthdate,
    }).toMatchSnapshot();
  });
});
