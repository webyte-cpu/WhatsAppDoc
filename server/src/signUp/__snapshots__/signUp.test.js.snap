// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up Has an invalid email and password 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [UserInputError: Variable "$email" got invalid value ""; Expected type "EmailAddress". Value is not a valid email address: ],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Sign Up Has valid email but invalid password 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Password format is not valid; Password must be 8 or more characters long and contain at least one of the following:
                  number, uppercase letter, lowercase letter, and symbol.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Sign Up Should sign up a Doctor 1`] = `
Object {
  "birthdate": undefined,
  "email": "doc_kyle@webyte.org",
  "firstName": "Kyle",
  "img": undefined,
  "lastName": "Osunero",
  "role": "DOCTOR",
  "sex": undefined,
}
`;

exports[`Sign Up Should sign up a Patient 1`] = `
Object {
  "birthdate": undefined,
  "email": "patient_kyle@webyte.org",
  "firstName": "Kyle",
  "img": undefined,
  "lastName": "Osunero",
  "role": "PATIENT",
  "sex": undefined,
}
`;

exports[`Sign Up Trys to sign up with existing user 1`] = `
Object {
  "birthdate": undefined,
  "email": "kent@webyte.org",
  "firstName": "Kent",
  "img": undefined,
  "lastName": "Handumon",
  "role": "PATIENT",
  "sex": undefined,
}
`;
